.github/workflows/publish-docs.yml

```diff
@@ -0,0 +1,35 @@
+name: Publish
+
+on:
+  push:
+    branches:
+        - kp-add-sphinx
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Install dependencies
+      run: |
+        export PATH="$HOME/.local/bin:$PATH"
+        sudo apt-get install -y python3-setuptools
+        pip3 install --user -r docs/requirements.txt
+    - name: Build the docs
+      run: |
+        export PATH="$HOME/.local/bin:$PATH"
+        make html
+    - name: Deploy the docs
+      run: |
+        mkdir $HOME/output
+        mv -v _build/html/* $HOME/output
+        touch $HOME/output/.nojekyll
+        cd $HOME/output
+        git init
+        git config --global user.name "${GITHUB_ACTOR}"
+        git config --global user.email "${GITHUB_ACTOR}@github.com"
+        git add .
+        git commit -m "latest html output"
```

Comment:
I think the answer is no. Since the process is automated there's no way to resolve conflicts, which is why it is recommended to force push the newly initialized HTML content.

Reply:
Force-pushing to `gh-pages` is fine (there is no need to resolve conflicts). To rephrase my original comment: I just would like the documentation build action to force push <the-content-of-source-branch-master> + <-the-latest-generated-documentation-from-this-source-branch>.

So IOW, instead of those `make html` and `git init + add + commit` steps, I'd like to have something like:
```
  - name: Branch off base for updated gh-pages
    run: |
      git branch gh-pages "${GITHUB_REF}"
      git checkout gh-pages
  - name: Add latest generated docs to gh-pages
    run: |
      export PATH="$HOME/.local/bin:$PATH"
      make html
      for f in $(find _build/html -type f); do \
          rm -f ./${f#_build/html/}; \
      done
      cp -rv _build/html/* .
      rm -fr html
      touch .nojekyll
      git add html .nojekyll
      git config --global user.name "${GITHUB_ACTOR}"
      git config --global user.email "${GITHUB_ACTOR}@github.com"
      git commit -m "publish-docs.yml: generated latest html output."
  - name: Publish/force-push latest gh-pages
    run: |
       git push -f https://${GITHUB_ACTOR}:${{secrets.ACCESS_TOKEN}}@github.com/${GITHUB_REPOSITORY}.git HEAD:gh-pages
```

Reply:
Ah. Gotcha. Should be fine. I'll take a look.

Reply:
I got the whole branch-generate-docs-add-push chain working with this:
```
jobs:
  update-gh-pages:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v1
    - name: Install dependencies
      run: |
        export PATH="$HOME/.local/bin:$PATH"
        sudo apt-get install -y python3-setuptools
        pip3 install --user -r docs/requirements.txt
    - name: Branch off base for updated gh-pages
      run: |
        git branch gh-pages
        git checkout gh-pages
    - name: Add latest generated docs to gh-pages
      shell: bash
      run: |
        PATH="$HOME/.local/bin:$PATH" make html
        cp -rv _build/html/* .
        for f in $(find _build/html -type f -name '.*'); do \
            cp -v $f ./${f#_build/html/}; \
        done
        for f in $(find _build/html -type f); do \
            f="${f#_build/html/}"; \
            echo "adding generated ${f#_build/html/} to git repo..."; \
            git add ./${f#_build/html/}; \
        done
        touch .nojekyll
        git add .nojekyll
        git config --global user.name "${GITHUB_ACTOR}"
        git config --global user.email "${GITHUB_ACTOR}@github.com"
        git commit -m "publish-docs.yml: generated latest html output." .
    - name: Publish/force-push to gh-pages
      shell: bash
      env:
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      run: |
        git push -f https://${GITHUB_ACTOR}:${GITHUB_TOKEN}@github.com/${GITHUB_REPOSITORY}.git HEAD:gh-pages
```

Reply:
OK. I'll copy/paste.